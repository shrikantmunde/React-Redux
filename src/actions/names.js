import { ADD_NAME, REMOVE_NAME, COPY_NAMES } from '../actions'

// export const addName = (name) => {
//   return {
//     type: ADD_NAME,
//     name
//   }
// }

// This is an action creatore
// It returns a simple object that describes the change you
// want to make to the state.
// All the actions must implement the "type" property
// Use a unique ID for the type
// Most people use a string constant
export function addName(name) {
  return {
    type: ADD_NAME,
    name : name
  }
}

export function removeName(index) {
  return {
    type: REMOVE_NAME,
    index 
  }
}

export function copyNames() {
  return {
    type: COPY_NAMES
  }
}
