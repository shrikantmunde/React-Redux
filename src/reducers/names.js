import { ADD_NAME, REMOVE_NAME, COPY_NAMES } from '../actions'

// const reducer = (state = [], action) => {
//   switch (action.type) {
//     case ADD_NAME:
//       return [ ...state, action.name ]
//     default:
//       return state
//   }
// }

function names(state = ["Shrikant"], action) {
  let copy

  switch (action.type) {
    case ADD_NAME:
      // Array spread operator
      // Copies an array (...state)
      // Appends new item
      return [ ...state, action.name ]
    case REMOVE_NAME:
      copy = [...state]
      copy.splice(action.index, 1)
      return copy
    case COPY_NAMES:
      return [...state, ...state]
    default:
      return state
  }
}

export default names
