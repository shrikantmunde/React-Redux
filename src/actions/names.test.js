import { removeName, copyNames } from './names'

it('creates the REMOVE_NAME action', () => {
  expect(removeName(2)).toMatchSnapshot()
})

it('creates the COPY_NAME action', () => {
  expect(copyNames()).toMatchSnapshot()
})
