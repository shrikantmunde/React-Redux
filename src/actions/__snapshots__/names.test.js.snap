exports[`test creates the COPY_NAME action 1`] = `
Object {
  "type": "COPY_NAMES",
}
`;

exports[`test creates the REMOVE_NAME action 1`] = `
Object {
  "index": 2,
  "type": "REMOVE_NAME",
}
`;
